app/models/miq_server.rb

```diff
@@ -24,29 +26,15 @@ class MiqServer < ApplicationRecord
   has_many                :messages,  :as => :handler, :class_name => 'MiqQueue'
   has_many                :miq_events, :as => :target, :dependent => :destroy
 
-  cattr_accessor          :my_guid_cache
-
   before_destroy          :validate_is_deleteable
 
   default_value_for       :rhn_mirror, false
 
   virtual_column :zone_description, :type => :string
 
-  RUN_AT_STARTUP  = %w( MiqRegion MiqWorker MiqQueue MiqReportResult )
-
-  STATUS_STARTING       = 'starting'.freeze
-  STATUS_STARTED        = 'started'.freeze
-  STATUS_RESTARTING     = 'restarting'.freeze
-  STATUS_STOPPED        = 'stopped'.freeze
-  STATUS_QUIESCE        = 'quiesce'.freeze
-  STATUS_NOT_RESPONDING = 'not responding'.freeze
-  STATUS_KILLED         = 'killed'.freeze
-
-  STATUSES_STOPPED = [STATUS_STOPPED, STATUS_KILLED]
-  STATUSES_ACTIVE  = [STATUS_STARTING, STATUS_STARTED]
-  STATUSES_ALIVE   = STATUSES_ACTIVE + [STATUS_RESTARTING, STATUS_QUIESCE]
-
-  RESTART_EXIT_STATUS = 123
+  include MiqServerBaseConstants
+  include MiqServerBaseMethods
+  include MiqServerQueueManagement
```

Comment:
These should be `MiqServer::BaseConstants`, etc. based on your file structure and that there are other modules already using that pattern. Also, the requires at the top probably won't be needed then as autoload should pick them up at the correct location.